src/Controllers/FilesController.js

```diff
@@ -26,7 +26,11 @@ export class FilesController extends AdaptableController {
       contentType = mime.lookup(filename);
     }
 
-    filename = randomHexString(32) + '_' + filename;
+    // increased the random string length to 36 so that it is the same length as an UUID with dashes.
+    // This makes file name property extraction for display in clients such as Parse Dashboard to work correctly.
+    // We can not use an UUID with dashes here because it is already used for files.parse.com hosted files and
+    // expandFilesInObject method below uses this distinction to construct the correct url for the file.
+    filename = randomHexString(36) + '_' + filename;
```

Comment:
That's true. So I can do something like this:

var hexstring = randomHexString(32)
var uuid = hexstring.substring(0,8) + "-" + hexstring.substring(8,12) + "-" + hexstring.substring(12,16) + "-" + hexstring.substring(16,20) + "-" + hexstring.substring(20);

filename = uuid + '_' + filename;